gmso/utils/misc.py

```diff
@@ -1,29 +1,31 @@
+from functools import lru_cache
 import unyt as u
 from unyt.exceptions import UnitConversionError
 
 
+#UNYT_CACHE = dict()
+
 def unyt_to_hashable(unyt_or_unyt_iter):
     """Convert a (list of) unyt array or quantity to a hashable tuple."""
     if unyt_or_unyt_iter is None:
         return unyt_or_unyt_iter
-    hash_coll = []
+
     if isinstance(unyt_or_unyt_iter, list):
-        for val in unyt_or_unyt_iter:
-            hash_coll.append(_unyt_to_hashable_single(val))
-        return tuple(hash_coll)
+        hash_coll = tuple(_unyt_to_hashable_single(val) for val in unyt_or_unyt_iter)
```

Comment:
For future developers sake, can we rename `hash_coll` to `hash_collision` or `collision`? If someone is less familiar with hashing and hashmaps, this might help.

Reply:
Is this supposed to mean `collection` rather than `collision`?

Reply:
I think you are correct, @umesh-timalsina. In that case, `collection` definitely makes more sense to me.